### ISSUE TITLE
Enhancement: Allow prediction on numpy arrays present as local variable

### ISSUE THREAD

**Issue by vrutikrabadia** [DevoLearn/devolearn#47]
The current model only allows us to predict on images by providing a path. We can extend the functionality and flexibility of out module by allowing numpy arrays as arguments to predict.

**Comment by Mayukhdeb** [comment]
You're right, I'm planning to add all of these functionalities through the `InferenceEngine` class [found here](https://github.com/DevoLearn/devolearn/blob/master/devolearn/base_inference_engine.py)

So a more sensible thing to do now is to modify all of the current models so that they're  a subclass of the `InferenceEngine` class.

**Comment by vrutikrabadia** [comment]
Okay

**Comment by vrutikrabadia** [comment]
So first we can do that and then continue with this.

**Comment by Mayukhdeb** [comment]
yes
